doc/source/how-to-configure-clients.rst

```diff
@@ -29,115 +28,151 @@ them back to Python dictionaries.
 .. note::
 
     Currently, there is no support for directly sending collection types (e.g., ``Set``,
-    ``List``, ``Map``) as values in configuration dictionaries. There are several
-    workarounds to send collections as values by converting them to one of the supported
-    value types (and converting them back on the client-side).
+    ``List``, ``Map``) as values in configuration dictionaries. To send collections,
+    convert them to a supported type (e.g., JSON string) and decode on the client side.
 
-    One can, for example, convert a list of floating-point numbers to a JSON string,
-    then send the JSON string using the configuration dictionary, and then convert the
-    JSON string back to a list of floating-point numbers on the client.
+    Example:
 
-Configuration through built-in strategies
------------------------------------------
+    .. code-block:: python
 
-The easiest way to send configuration values to clients is to use a built-in strategy
-like ``FedAvg``. Built-in strategies support so-called configuration functions. A
-configuration function is a function that the built-in strategy calls to get the
-configuration dictionary for the current round. It then forwards the configuration
-dictionary to all the clients selected during that round.
+        import json
 
-Let's start with a simple example. Imagine we want to send (a) the batch size that the
-client should use, (b) the current global round of federated learning, and (c) the
-number of epochs to train on the client-side. Our configuration function could look like
-this:
+        # On the server
+        config_dict = {"data_splits": json.dumps([0.8, 0.1, 0.1])}
+
+        # On the client
+        data_splits = json.loads(config["data_splits"])
+
+Using Built-in Strategies for Configuration
+-------------------------------------------
+
+Flower supports configuration functions to dynamically adjust parameters sent to
+clients. Built-in strategies like ``FedAvg`` allow for setting configuration values for
+each round via a function.
+
+Example: Sending Training Configurations
```

Comment:
I think it's worth mentioning somewhere in this section that a similar process needs to be followed for `on_evaluate_config_fn`

Reply:
Currently mentioning it between line 123-141: 

```python

    def evaluate_config(server_round: int):
        """Generate evaluation configuration for each round."""
        # Create the configuration dictionary
        config = {
            "batch_size": 64,
            "current_round": server_round,
            "metrics": ["accuracy"],  # Example metrics to compute
        }
        return config
```

You can pass this `evaluate_config` callback to a built-in strategy like `FedAvg`:

```python

    strategy = FedAvg(
        on_evaluate_config_fn=evaluate_config  # Assign the evaluate_config function
    )

```